Keep ICU data-file dependencies only when using an external data file.

Origin: EmberBSD contributors (AI-assisted); upstream status: not submitted.

--- components/url_formatter/spoof_checks/top_domains/BUILD.gn
+++ components/url_formatter/spoof_checks/top_domains/BUILD.gn
@@ -3,6 +3,7 @@
 # found in the LICENSE file.
 
 import("//build/compiled_action.gni")
+import("//third_party/icu/config.gni")
 
 if (!is_ios && !is_android) {
   executable("make_top_domain_skeletons") {
@@ -45,11 +46,16 @@
     "//components/url_formatter/spoof_checks/top_domains/domains.skeletons",
     "//components/url_formatter/spoof_checks/top_domains/top_domains_trie_header.template",
   ]
-  inputs = _tool_inputs + [ "$root_build_dir/icudtl.dat" ]
+  inputs = _tool_inputs
+  if (icu_use_data_file) {
+    inputs += [ "$root_build_dir/icudtl.dat" ]
+  }
   outputs = [ "$target_gen_dir/domains-trie.h" ]
   args = rebase_path(_tool_inputs, root_build_dir) +
          rebase_path(outputs, root_build_dir)
-  deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  if (icu_use_data_file) {
+    deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  }
 }
 
 # TODO(crbug.com/41432246): Combine this and the previous one into a
@@ -62,12 +68,17 @@
     "//components/url_formatter/spoof_checks/top_domains/idn_test_domains.skeletons",
     "//components/url_formatter/spoof_checks/top_domains/idn_test_top_domains_trie_header.template",
   ]
-  inputs = _tool_inputs + [ "$root_build_dir/icudtl.dat" ]
+  inputs = _tool_inputs
+  if (icu_use_data_file) {
+    inputs += [ "$root_build_dir/icudtl.dat" ]
+  }
   outputs = [ "$target_gen_dir/idn_test_domains_trie.h" ]
 
   args = rebase_path(_tool_inputs, root_build_dir) +
          rebase_path(outputs, root_build_dir)
-  deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  if (icu_use_data_file) {
+    deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  }
 }
 
 # TODO(crbug.com/41432246): Combine this and the previous one into a
@@ -80,12 +91,17 @@
     "//components/url_formatter/spoof_checks/top_domains/test_domains.skeletons",
     "//components/url_formatter/spoof_checks/top_domains/test_top_domains_trie_header.template",
   ]
-  inputs = _tool_inputs + [ "$root_build_dir/icudtl.dat" ]
+  inputs = _tool_inputs
+  if (icu_use_data_file) {
+    inputs += [ "$root_build_dir/icudtl.dat" ]
+  }
   outputs = [ "$target_gen_dir/test_domains_trie.h" ]
 
   args = rebase_path(_tool_inputs, root_build_dir) +
          rebase_path(outputs, root_build_dir)
-  deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  if (icu_use_data_file) {
+    deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  }
 }
 
 executable("make_top_domain_list_variables") {
@@ -131,11 +147,16 @@
   # Inputs in order expected by the command line of the tool.
   _tool_inputs =
       [ "//components/url_formatter/spoof_checks/top_domains/domains.list" ]
-  inputs = _tool_inputs + [ "$root_build_dir/icudtl.dat" ]
+  inputs = _tool_inputs
+  if (icu_use_data_file) {
+    inputs += [ "$root_build_dir/icudtl.dat" ]
+  }
   outputs = [ "$target_gen_dir/top-bucket-domains-inc.cc" ]
   args = rebase_path(_tool_inputs, root_build_dir) + [ "top_bucket_domains" ] +
          rebase_path(outputs, root_build_dir)
-  deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  if (icu_use_data_file) {
+    deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  }
 }
 
 # TODO(crbug.com/41432246): Combine this and the previous one into a
@@ -147,11 +168,16 @@
   _tool_inputs = [
     "//components/url_formatter/spoof_checks/top_domains/test_domains.list",
   ]
-  inputs = _tool_inputs + [ "$root_build_dir/icudtl.dat" ]
+  inputs = _tool_inputs
+  if (icu_use_data_file) {
+    inputs += [ "$root_build_dir/icudtl.dat" ]
+  }
   outputs = [ "$target_gen_dir/test-top-bucket-domains-inc.cc" ]
   args = rebase_path(_tool_inputs, root_build_dir) +
          [ "test_top_bucket_domains" ] + rebase_path(outputs, root_build_dir)
-  deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  if (icu_use_data_file) {
+    deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  }
 }
 
 # top_bucket_domains and top_bucket_domains_header are intentionally separated
--- third_party/blink/renderer/platform/BUILD.gn
+++ third_party/blink/renderer/platform/BUILD.gn
@@ -5,6 +5,7 @@
 import("//base/trace_event/tracing.gni")
 import("//build/buildflag_header.gni")
 import("//build/compiled_action.gni")
+import("//third_party/icu/config.gni")
 import("//build/config/compiler/compiler.gni")
 import("//build/config/ui.gni")
 import("//build/nocompile.gni")
@@ -132,13 +133,17 @@
 
 compiled_action("character_data") {
   tool = ":character_data_generator"
-  inputs = [ "$root_build_dir/icudtl.dat" ]
+  if (icu_use_data_file) {
+    inputs = [ "$root_build_dir/icudtl.dat" ]
+  }
   outputs = [
     "$blink_platform_output_dir/character_property_data.cc",
     "$blink_platform_output_dir/text/break_iterator_data_inline_header.h",
   ]
   args = rebase_path(outputs, root_build_dir)
-  deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  if (icu_use_data_file) {
+    deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  }
 }
 
 executable("character_data_generator") {
--- components/autofill/core/browser/BUILD.gn
+++ components/autofill/core/browser/BUILD.gn
@@ -4,6 +4,7 @@
 
 import("//build/buildflag_header.gni")
 import("//build/compiled_action.gni")
+import("//third_party/icu/config.gni")
 import("//build/config/chrome_build.gni")
 import("//build/config/features.gni")
 import("//build/config/python.gni")
@@ -487,12 +488,16 @@
 
 compiled_action("generate_country_native_names") {
   tool = ":country_native_names_generator"
-  inputs = [ "$root_build_dir/icudtl.dat" ]
+  if (icu_use_data_file) {
+    inputs = [ "$root_build_dir/icudtl.dat" ]
+  }
   outputs = [ "$target_gen_dir/geo/country_native_names_inl.h" ]
   args = [ "--output-file-path=" +
            rebase_path("$target_gen_dir/geo/country_native_names_inl.h",
                        root_build_dir) ]
-  deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  if (icu_use_data_file) {
+    deps = [ "//third_party/icu:copy_icudata($default_toolchain)" ]
+  }
 }
 
 proto_library("amount_extraction_heuristic_regexes_proto") {
